src/desdemona/query.clj

```diff
@@ -0,0 +1,25 @@
+(ns desdemona.query
+  (:require [clojure.core.logic :as l]))
+
+(defn ^:private gen-query
+  "Expands a query and events to a core.logic program that executes
+  it."
+  [n-answers query events]
+  `(l/run ~n-answers [results#]
+     (l/fresh [~'x]
+       (l/== [~'x] results#)
+       (l/membero ~'x ~events)
+       ~query)))
```

Comment:
OK, cool. I'll try and think of a good way of wording that; it's just syntax-quote because that's easy syntax for a bunch of lists with some stuff interpolated I guess?